[ci] Fix broken workflows

Kudo opened the pull request:
# Why

fix ci broken

# How

the broken on ios relates to cocoapods with `incremental_installation` which `installer.pod_project` might be nil if no new project generated.

the broken on android:
  - [detox] upgrade detox to 18.20.1 to support RN 0.64
  - [detox] allow local server cleartext traffic that i disallowed from release build during RN upgrade
  - [detox] fix building error with react-native-screens
  - [react-native] cherry-pick upstream patch which prevent *.so not packed in aar.

# Test Plan

ci passed

Kudo committed:
[ci] try to fix ios test-suite breaks

Kudo committed:
[bare-expo][android] fix detox testing

 - upgrade detox to 18.20.1 to support rn 0.64
 - allow local server cleartext traffic that i disallow from release
   build during rn upgrade

tsapeta commented on .github/workflows/test-suite.yml:
```diff
@@ -103,9 +103,9 @@ jobs:
         with:
           path: 'apps/bare-expo/ios/Pods'
-          key: ${{ runner.os }}-bare-expo-pods-${{ hashFiles('apps/bare-expo/ios/Podfile.lock') }}
+          key: ${{ runner.os }}-bare-expo-pods-v2-${{ hashFiles('apps/bare-expo/ios/Podfile.lock') }}
```
If you're sure it's cache issue, I would just wait for it to be invalidated instead of changing the keys 😉 However, I wonder why because the lock has changed after upgrade 🤔

Kudo replied:
the root cause is `installer.pods_project` be null. according to [this](https://github.com/CocoaPods/CocoaPods/issues/8604#issuecomment-472075100), bare-expo enabled `incremental_installation` make it possibly happened. that's why i tried not to use old cache.

do you know how to invalidate cache? according the [GH document](https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy), we should wait 7 days.

btw GH actions is [abnormal now](https://www.githubstatus.com/incidents/7p1nnvkgh96y), waiting for they fixing it.

Kudo replied:
can reproduce the error locally by `pod install` twice. (which `incremental_installation` takes effect)
so i reverted the cache key change and further check if `installer.pod_project` is not nil.

tsapeta replied:
Afaik there is no way to immediately invalidate the cache, but I meant that any change in the lock should make it work 🙂 Also, there is a limit of 5GB for caches per repository, which I believe we achieve pretty quickly so older caches are for sure deleted in a few days.

Kudo committed:
[jest-expo] Update transformIgnorePatterns after RN 0.64 upgrade

Upstream change:
https://github.com/facebook/react-native/commit/a77f2c40d1044da63426078f356cfc7c1284a79e

Kudo committed:
[html-elements] rebuild to reflect react-native types change

Kudo committed:
add missing comma

brentvatne commented on packages/jest-expo/jest-preset.js:
```diff
@@ -38,7 +38,7 @@ if (!Array.isArray(jestPreset.transformIgnorePatterns)) {
 // Also please keep `testing-with-jest.md` file up to date
 jestPreset.transformIgnorePatterns = [
-  'node_modules/(?!(jest-)?react-native|@react-native-community|expo(nent)?|@expo(nent)?/.*|react-navigation|@react-navigation/.*|@unimodules/.*|unimodules|sentry-expo|native-base|react-native-svg)',
+  'node_modules/(?!((jest-)?react-native|@react-native(-community)?)|expo(nent)?|@expo(nent)?/.*|react-navigation|@react-navigation/.*|@unimodules/.*|unimodules|sentry-expo|native-base|react-native-svg)',
```
i imagine the purpose of this is to handle https://www.npmjs.com/org/react-native :+1:

Kudo replied:
cpojer published some `@react-native` prefixed packages and landed on RN 0.64, e.g. `@react-native/polyfiils`

Kudo committed:
Revert "[ci] try to fix ios test-suite breaks"

This reverts commit c794f6682f548d1730369d4a90bee919ab563a01.

Kudo committed:
[bare-expo] Fix pod install failure from incremental_installation

Kudo committed:
[react-native] upgrade with the patch to fix *.so occasionally not packed into aar

Kudo committed:
fix building error with detox + react-native-screens

https://github.com/wix/Detox/issues/2848#issuecomment-874803711

Kudo committed:
add comment [skip ci]
